src/libasr/pass/pass_utils.h

```diff
@@ -578,6 +584,32 @@ namespace LCompilers {
             bool perform_cast=false,
             ASR::cast_kindType cast_kind=ASR::cast_kindType::IntegerToInteger,
             ASR::ttype_t* casted_type=nullptr) {
+            if ( x->m_dt_sym && ASR::is_a<ASR::Struct_t>(*(x->m_dt_sym)) ) {
```

Comment:
```suggestion
            if ( ASR::is_a<ASR::Struct_t>(*(x->m_dt_sym)) ) {
```
Then check if the Struct scope has a __init__ function, if not jump to previous implementation. If yes, replace with a Subroutine Call as you did here.